src/paradigma/heart_rate/tfd.py

```diff
@@ -0,0 +1,553 @@
+        win_param = kern_params['win_param'] if l >= 3 else 0
+        win_param2 = kern_params['win_param2'] if l >= 4 else 1
+
+        G1 = get_window(win_length, win_type, win_param)
+        G1 = pad_window(G1, N)
+
+        # Define window in the time domain or Doppler domain
+        if win_param2 == 0:
+            G1 = np.fft.fft(G1)
+            G1 /= G1[0]
+
+        g[:] = G1 # Assign the window to the kernel
+
+    elif kern_type == 'pwvd':
+        key = 'doppler'
+        # Pseudo-Wigner-Ville Distribution (Doppler Independent kernel)
+        if l < 2:
+            raise ValueError("Need at least two window parameters for PWVD") 
+        win_length = kern_params['win_length']
+        win_type = kern_params['win_type']
+        win_param = kern_params['win_param'] if l >= 3 else 0
+        win_param2 = kern_params['win_param2'] if l >= 4 else 0
+
+        G2 = get_window(win_length, win_type, win_param) # Generate the window, same per lag iteration
+        G2 = pad_window(G2, N)      # Zero-pad the window to the length of the signal
+        G2 = G2[lag_index]          # Extract the lag_index-th element of the window --> can this be done more efficiently? Since we only need one element of the window and the padding is similar for every iteration
+
+        g[:] = G2 # Assign the window to the kernel
+
+    elif kern_type == 'sep':
+        # Separable Kernel
+        g1 = np.copy(g)  # Create a new array for g1
+        g2 = np.copy(g)  # Create a new array for g2
+        
+        # Call recursively to obtain g1 and g2 kernels (no in-place modification of g)
+        g1 = get_kern(g1, lag_index, 'swvd', kern_params['lag'], N) # Generate the first kernel
+        g2 = get_kern(g2, lag_index, 'pwvd', kern_params['doppler'], N) # Generate the second kernel
+        g = g1 * g2 # Multiply the two kernels to obtain the separable kernel
+
+    else:
+        raise ValueError(f"Unknown kernel type: {kern_type}")
+
+    return g
+
+def get_window(win_length, win_type, win_param=None, dft_window=False, Npad=0):
+    """
+    General function to calculate a window function.
+    
+    Parameters:
+    -----------
+    win_length : int
+        Length of the window.
+    win_type : str
+        Type of window. Options are:
+        {'delta', 'rect', 'hamm'|'hamming', 'hann'|'hanning', 'gauss', 'cosh'}.
+    win_param : float, optional
+        Window parameter (e.g., alpha for Gaussian window). Default is None.
+    dft_window : bool, optional
+        If True, returns the DFT of the window. Default is False.
+    Npad : int, optional
+        If greater than 0, zero-pads the window to length Npad. Default is 0.
+    
+    Returns:
+    --------
+    win : ndarray
+        The calculated window (or its DFT if dft_window is True).
+    """
+    # Handle optional arguments
+    if win_param is None:
+        win_param = []
+    
+    # Get the window
+    win = get_win(win_length, win_type, win_param, dft_window)
```

Comment:
I see your `get_win` function here assigns an empty list to `win_param`. However, in the `get_win()` function you assign floats to this variable. Is this working as expected?

Reply:
see previous comment